shell/browser/api/electron_api_desktop_capturer.cc

```diff
@@ -28,8 +29,110 @@
 #include "third_party/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h"
 #include "third_party/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h"
 #include "ui/display/win/display_info.h"
+#elif BUILDFLAG(IS_LINUX)
+#if defined(USE_OZONE_PLATFORM_X11)
+#include "base/logging.h"
+#include "ui/base/x/x11_display_util.h"
+#include "ui/base/x/x11_util.h"
+#include "ui/display/util/display_util.h"
+#include "ui/display/util/edid_parser.h"
+#include "ui/gfx/x/randr.h"
+#include "ui/gfx/x/x11_atom_cache.h"
+#include "ui/gfx/x/xproto_util.h"
+#endif  // defined(USE_OZONE_PLATFORM_X11)
 #endif  // BUILDFLAG(IS_WIN)
 
+#if BUILDFLAG(IS_LINUX)
+// Find the mapping between output identifier and the monitor name atom
+// Note this isn't the atom string, but the numeric atom identifier,
+// since this is what the WebRTC system uses as the display identifier
+// Based on `GetMonitors` in ui/base/x/x11_display_util.cc
+std::map<x11::RandR::Output, int32_t> GetMonitorsNameAtom(int version,
+                                                          x11::RandR* randr,
+                                                          x11::Window window) {
+  std::map<x11::RandR::Output, int32_t> output_to_monitor;
+  if (version >= 105) {
+    if (auto reply = randr->GetMonitors({window}).Sync()) {
+      for (size_t monitor = 0; monitor < reply->monitors.size(); monitor++) {
+        for (x11::RandR::Output output : reply->monitors[monitor].outputs)
+          output_to_monitor[output] =
+              static_cast<int32_t>(reply->monitors[monitor].name);
+      }
+    }
+  }
+  return output_to_monitor;
+}
```

Comment:
The key thing is getting the `name` field from the reply for the monitor, since that . I can change it to use the unmodified upstream function, but at the cost of making another call to `randr.GetMonitors`, to then be able to get the name atom for a given monitor. I figured that since `GetMonitors` is a private function that would need to be copied in anyway, I may as well make it slightly more ergonomic for the intended purpose.  I see the logic about staying closer to upstream to make it easier to absorb their changes though; I can change this to do that second look-up separately.